sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -191,11 +195,18 @@ private[sql] class SharedState(
 }
 
 object SharedState extends Logging {
-  try {
-    URL.setURLStreamHandlerFactory(new FsUrlStreamHandlerFactory())
-  } catch {
-    case e: Error =>
-      logWarning("URL.setURLStreamHandlerFactory failed to set FsUrlStreamHandlerFactory")
+  private val fsUrlStreamHandlerFactoryInitialized = new AtomicBoolean(false)
+
+  private def setFsUrlStreamHandlerFactory(conf: SparkConf): Unit = {
+    if (conf.get(DEFAULT_URL_STREAM_HANDLER_FACTORY_ENABLED) &&
+        fsUrlStreamHandlerFactoryInitialized.compareAndSet(false, true)) {
+      try {
+        URL.setURLStreamHandlerFactory(new FsUrlStreamHandlerFactory())
+      } catch {
+        case NonFatal(_) =>
```

Comment:
Meh, maybe worth logging the exception?

Reply:
It's a behavior change which I don't want to introduce.

Reply:
In this PR, I added that, but I reverted it at https://github.com/apache/spark/pull/26530/commits/d9e0ec73dc6a9685b65a08c8166129299f6c8960 due to the verbose error message (which is different from the previous Spark behavior).

Reply:
OK, no big deal. It's just a log message in either event.